homeassistant/components/overkiz/water_heater_entities/domestic_hot_water_production.py

```diff
@@ -0,0 +1,281 @@
+                            operating_mode.get(OverkizCommandParam.RELAUNCH),
+                        )
+                        == OverkizCommandParam.ON
+                    )
+                return False
+
+            return cast(str, operating_mode) in DHWP_BOOST_MODES
+
+        return False
+
+    @property
+    def is_away_mode_on(self) -> bool | None:
+        """Return true if away mode is on."""
+
+        if self.executor.has_state(OverkizState.IO_DHW_ABSENCE_MODE):
+            return (
+                self.executor.select_state(OverkizState.IO_DHW_ABSENCE_MODE)
+                == OverkizCommandParam.ON
+            )
+
+        if self.executor.has_state(OverkizState.MODBUSLINK_DHW_ABSENCE_MODE):
+            return (
+                self.executor.select_state(OverkizState.MODBUSLINK_DHW_ABSENCE_MODE)
+                == OverkizCommandParam.ON
+            )
+
+        operating_mode = self.executor.select_state(OverkizState.CORE_OPERATING_MODE)
+
+        if operating_mode:
+            if isinstance(operating_mode, dict):
+                if operating_mode.get(OverkizCommandParam.ABSENCE):
+                    return (
+                        cast(
+                            str,
+                            operating_mode.get(OverkizCommandParam.ABSENCE),
+                        )
+                        == OverkizCommandParam.ON
+                    )
+                if operating_mode.get(OverkizCommandParam.AWAY):
+                    return (
+                        cast(
+                            str,
+                            operating_mode.get(OverkizCommandParam.AWAY),
+                        )
+                        == OverkizCommandParam.ON
+                    )
+                return False
+
+            return cast(str, operating_mode) in DHWP_AWAY_MODES
+
+        return None
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.CORE_MINIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.CORE_MAXIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.IO_MIDDLE_WATER_TEMPERATURE,
+                OverkizState.MODBUSLINK_MIDDLE_WATER_TEMPERATURE,
+            ),
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        return cast(
+            float, self.executor.select_state(OverkizState.CORE_TARGET_TEMPERATURE)
+        )
```

Comment:
Should we also do something like 
```python
        target_temperature = self.device.states[
            OverkizState.MODBUS_CONTROL_DHW_SETTING_TEMPERATURE
        ]
        if target_temperature:
            return target_temperature.value_as_float
        return None
```
from #81536

Reply:
Done there and in all other places.  
For max and min temp, I had to add a cast because value_as_float could return None and the water heater required a float